cache/cache.go

```diff
@@ -62,7 +62,8 @@ type Config struct {
 	// Cost evaluates a value and outputs a corresponding cost. This function
 	// is ran after Set is called for a new item or an item update with a cost
 	// param of 0.
-	Cost func(value interface{}) int64
+	OnInsert func(key uint64, cost int64)
+	Cost     func(value interface{}) int64
```

Comment:
```suggestion
	Cost     func(value interface{}) int64
	OnInsert func(key uint64, cost int64)
```

Swap lines to match comments